### DISCUSSION TITLE
Rdkit JS (MinimalLib)

### DISCUSSION THREAD

**Discussion by mgedgaudas** [rdkit/rdkit#4661] [Q&A]
Is it possible to get molecules molecular formula (benzene => C6H6) using the javascript/wasm implementation of rdkit?
if yes, how?

**Comment by ptosco** [comment]
@mgedgaudas Not at the moment. However, it is simple to wrap functionality which already exists in the C++ layer.
For example, I have attached a patch file that adds a `JSMol::get_formula` function to do that.
If you `git clone` the current RDKit `master` branch, apply the patch
```
gunzip -c get_formula.diff.gz | patch -p1
```
and rebuild the RDKitJS library using the `Dockerfile` in `Code/MinimalLib/docker`, you may then
 use the additional functionality that you need.

Here's the patch: [get_formula.diff.gz](https://github.com/rdkit/rdkit/files/7448373/get_formula.diff.gz)
